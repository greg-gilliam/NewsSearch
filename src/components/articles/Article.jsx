import React, { Component } from 'react';

const Article = ({ title, description, content }) => (
  <article>
    <p>{title}</p>
    <p>{description}</p>
    <p>{content}</p>
  </article>
);

export default Article;
