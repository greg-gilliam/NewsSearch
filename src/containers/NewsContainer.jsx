import React, { Component } from 'react';
import { fetchNews } from '../../services/NewsApi';
import ArticleList from '../components/articles/ArticleList';

class NewsContainer extends React.Component {
  state = {
    loading: true,
    news: [],
  };

  async componentDidMount() {
    const news = await fetchNews();
    this.setState({
      news,
      loading: false,
    });
  }

  render() {
    const { loading, news } = this.state;

    if (loading) {
      return <h1>Loading...</h1>;
    }
    return <ArticleList news={news} />;
  }
}

export default NewsContainer;
