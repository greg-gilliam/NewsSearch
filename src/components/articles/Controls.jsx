import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Article from './Article.jsx';

export default function Controls({ query, onChange, onSubmit }) {
  return (
    <form onSubmit={onSubmit}>
      <label htmlFor="inputField">Search for an article!</label>
      <input
        id="inputField"
        name="inputField"
        type="text"
        value={query}
        onChange={(event) => onChange(event)}
      />
      <button>Submit</button>
    </form>
  );
}

Controls.propTypes = {
  query: PropTypes.string,
  onChange: PropTypes.func,
  onSubmit: PropTypes.func,
};
