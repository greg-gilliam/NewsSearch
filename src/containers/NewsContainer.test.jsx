import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import NewsContainer from './NewsContainer';
import userEvent from '@testing-library/user-event';
import Article from '../components/articles/Article';
import Controls from '../components/articles/Controls';

describe('News Container', () => {
  it('Displays a list of articles', async () => {
    render(<NewsContainer />);
    screen.getAllByText('Loading...');

    const ul = await screen.findByRole('list', { name: 'searchResult' });
    expect(ul).toMatchSnapshot();

    const searchInput = await screen.findByLabelText('Search for an article!');

    const submitButton = await screen.findByRole('button', {
      name: 'get-news',
    });

    userEvent.click(submitButton);

    await waitFor(() => {
      const yourNews = screen.getAllByTitle('theTitle');
      expect(yourNews).toHaveLength(5);
    });
  });

  it('Displays an article', () => {
    const { asFragment } = render(<Article title="theTitle" />);
    expect(asFragment()).toMatchSnapshot();
  });

  // it('Should search', () => {
  //   const { asFragment } = render(
  //     <Controls onSubmit="news" query="news" onChange="news" />
  //   );
  //   expect(asFragment()).toMatchSnapshot();
  // });
});
