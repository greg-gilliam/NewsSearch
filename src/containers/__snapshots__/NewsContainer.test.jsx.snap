// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Container Displays a list of articles 1`] = `
<ul
  aria-label="searchResult"
>
  <li>
    <article>
      <p
        title="theTitle"
      >
        前三季度我国国际收支基本平衡--经济·科技 - 人民网
      </p>
      <p>
        本报北京11月5日电  （记者葛孟超）国家外汇管理局5日公布2021年前三季度国际收支平衡表初步数据。数据显示，2021年前三季度，我国国际收支基本平衡。其中，经常账户顺差20
      </p>
      <p />
    </article>
  </li>
  <li>
    <article>
      <p
        title="theTitle"
      >
        American Airlines jacks up flight attendant holiday pay to avoid more flight cancellations - CNBC
      </p>
      <p>
        American Airlines offers as much as triple-flight attendant pay for working holiday trips if they also have perfect attendance through Jan. 2.
      </p>
      <p>
        American Airlines, jolted by mass cancellations last weekend, has offered flight attendants as much as triple their pay to work holiday trips if they also have perfect attendance through early Januar… [+2461 chars]
      </p>
    </article>
  </li>
  <li>
    <article>
      <p
        title="theTitle"
      >
        Alphabet, cada vez más cerca del exclusivo club de 2 billones de dólares - El CEO
      </p>
      <p>
        Alphabet, propietaria de Google, está un paso de superar el valor de mercado de 2 billones de dólares, gracias a un aumento del 69% este año impulsado por un repunte en el gasto en anuncios digitales y el crecimiento de su negocio en la nube. Las acciones sub…
      </p>
      <p>
        5 de noviembre 2021 | 3:52 pm
Alphabet, propietaria de Google, está un paso de superar el valor de mercado de 2 billones de dólares, gracias a un aumento del 69% este año impulsado por un repunte en… [+1298 chars]
      </p>
    </article>
  </li>
  <li>
    <article>
      <p
        title="theTitle"
      >
        ¿Compraste acciones de Tesla hace 11 años? Tu dinero creció cinco cifras - Dinero en imagen
      </p>
      <p>
        Mientras que los que invirtieron en Tesla en el 2011, tendrán una rentabilidad de cinco cifras.
      </p>
      <p>
        En enero de 2010 Tesla presentó su oferta pública inicial, en ese momento era una empresa emergente de apenas 6 años, más conocida por su vehículo eléctrico Roadster, así que apostar por la empresa d… [+932 chars]
      </p>
    </article>
  </li>
  <li>
    <article>
      <p
        title="theTitle"
      >
        B.C. court rules Edward Rogers is legitimate chair of telecom company - CBC.ca
      </p>
      <p>
        A British Columbia court has ruled that Edward Rogers is the legitimate chair of Rogers Communications Inc., a major legal victory in his battle to take control of the company that bears his family name.
      </p>
      <p>
        A British Columbia court has ruled that Edward Rogers is the legitimate chair of Rogers Communications Inc., a major legal victory in his battle to take control of the company that bears his family n… [+2490 chars]
      </p>
    </article>
  </li>
</ul>
`;
